lib/board.rb

```diff
@@ -0,0 +1,52 @@
+class Board
+  attr_accessor :cells
+
+  def initialize
+    reset!
+  end
+
+  # Resets the state of the cells in the board to a 9 element array of " ".
+  def reset!
+    @cells = Array.new(9, " ")
+  end
+
+  # Prints the board.
+  def display
+    puts " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} "
+    puts "-----------"
+    puts " #{@cells[3]} | #{@cells[4]} | #{@cells[5]} "
+    puts "-----------"
+    puts " #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+    puts " "
+  end
+
+  # Takes in user input and returns the value of the board cell.
+  def position(position)
+    @cells[position.to_i-1]
+  end
+
+  # Returns true for a full board,returns false for an in-progress game.
+  def full?
+    !@cells.include?(" ")
```

Comment:
Use the most expressive and efficient iterator for a problem. `Board#full?` is asking if all elements are not empty strings. You can use logic that is symmetrical to those semantics with the [`#all?`](http://ruby-doc.org/core-2.2.3/Enumerable.html#method-i-all-3F) iterator.

`cells.all? {|cell| cell != " "}`

This is also more efficient in that the loop will stop after the iteration for which a cell is not empty.